DebuggableASTInterpreter/DASTClosure.class.st

```diff
@@ -34,19 +34,14 @@ DASTClosure >> body [
 	^ nodeAST body
 ]
 
-{ #category : #testing }
-DASTClosure >> doSemanticAnalysis [
-
-	^ nodeAST 
-]
-
 { #category : #initialization }
-DASTClosure >> initializeWith: aRBNode [ 
-	
+DASTClosure >> initializeWith: aRBNode [
+
 	sourceCode := aRBNode sourceCode.
 	nodeAST := aRBNode.
-	numArgs := nodeAST arguments size
-	
+	numArgs := nodeAST arguments size.
+
+	nodeAST doSemanticAnalysis
```

Comment:
why not `self nodeAST: aRBNode`?
Maybe I'm missing something.

Anyways, it could be merged like this and could be fixed later.
thoughts?

Reply:
You're right. We can call the setter and it would solve everything